packages/tool-cache/src/tool-cache.ts

```diff
@@ -200,7 +200,18 @@ export async function extractTar(
   }
 
   dest = dest || (await _createExtractFolder(dest))
-  const tarPath: string = await io.which('tar', true)
+  let tarPath: string
+  if (IS_WINDOWS) {
+    // Do not which tar on Windows due to incompatibilities between BSD tar and GNU tar on Windows.
+    // The version of tar at C:\Windows\System32\tar.exe is BSD tar and works properly with
+    // Windows-style paths. Whereas GNU tar requires forward slashes and also requires the flag
+    // `--force-local` due to the `:` in Windows paths. This presents an incompatibility because
+    // BSD tar does not support the flag `--force-local` (invalid flag, fails the process).
+    tarPath = path.join(process.env['windir'] as string, 'System32', 'tar.exe')
```

Comment:
Tar seems to have been added as a part of the initiative to bring more Unix tools to windows in Windows 10. It looks liked our only hosted version of windows is now 2019, and it exists on that machine for hosted builds

However, for self hosted runners do we know if this exists on all the platforms that can run .net core 3.0? Those go back all the way to Windows Server 2012. https://github.com/dotnet/core/blob/master/release-notes/3.0/3.0-supported-os.md

For 7zip, we ship a copy of the binary and and use it if needed, we may need to look into a similar solution here.
https://github.com/actions/toolkit/blob/d823f7598797eef5514afdc84f8ef51e293091f0/packages/tool-cache/src/tool-cache.ts#L153

Reply:
Should we attempt to detect if the path doesn't exist so we can log a better message in this case if not all windows versions will have it installed

Reply:
Looks like maybe using existsSync and handling the false case would be the best way to do this. I wouldn't attempt to ship a binary in this case... maybe just fail and explain why.

Reply:
+1 regarding the error message, at a glance it looked good to me but looking again we can make it a little bit more clear by try/catch/throw a better error.

one problem with the 7z approach is this function is literally designed as a wrapper over `tar` (see the `flags` parameter`)

Reply:
Agreed on using 7zip isn't the correct solution here. I don't really like shipping an exe of tar with the package like we do with 7zip either.

I'm happy with better error message (this may also affect users using container actions because iirc we support windows containers if the host system is linux)

Reply:
actually tool runner gives a good error already so i'll remove the `statSync` line

Reply:
looking into 7z. note we dont package the full 7z (over 1mb) just the decoder exe (40kb)

Reply:
i talked w/ josh and basically the cache action needs the support

Reply:
so will need broad os support.

i still think the current proposal is incrementally better

Reply:
Re: Hosted Runners vs Self-Hosted Runners
The current behavior is that we use GNU specific tar option on windows (specifically `--force-local`) since the GNU tar is prepended to the PATH.
This works for hosted runners, but self-hosted runners most likely do not have GNU prepended, so they will either fail with tar not being found or BSD tar not recognizing `--force-local`), see https://github.com/actions/cache/issues/91

With this solution, at least self-hosted runners that have tar are supported.

Also worth noting the cache action does not consume this package, it just has the same problems with tar.